File: vaadin-demo/frontend/animated-avatar.js
class AnimatedAvatar extends HTMLElement {
	constructor() {
		super();

		this.attachShadow({ mode: "open" });
		this.shadowRoot.innerHTML = `
					<style>
						html {
							background-color: darkslategrey;
							color: white;
						}

						span {
							font-style: italic;
						}

						.container {
							border: 1px solid darkgray;
							margin: 1em;
							padding: 1em;
						}

						.container img {
							width: 100%;
						}

						.animated {
							transform: rotate(360deg);
							transition-duration: 1s;
						}
					</style>
					<div class="container">
						<span id="span"></span>
						<img id = "img" src="">
						<slot id="footer"></slot>
					</div>
				`;

	}

	connectedCallback() {
		let img = this.shadowRoot.getElementById("img");
		img.onclick = () => this.animateElement(img);
		img.setAttribute("src", this.getAttribute("image"));

		let span = this.shadowRoot.getElementById("span");
		span.innerText = this.getAttribute("title");
	}

	animateElement(element) {
		element.className = "animated";
		element.addEventListener("transitionend", () => element.className = "");
	}
}

customElements.define("animated-avatar", AnimatedAvatar);

File: vaadin-demo/src/main/java/com/example/AnimatedAvatar.java
package com.example;

import com.vaadin.flow.component.Component;
import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;

@JsModule("./animated-avatar.js")
@Tag("animated-avatar")
public class AnimatedAvatar extends Component {

  public AnimatedAvatar(String title, String image, Component footer) {
    getElement().setAttribute("title", title);
    getElement().setAttribute("image", image);
    
    footer.getElement().setAttribute("name", "footer");
    getElement().appendChild(footer.getElement());
  }
  
  

}
